### PR TITLE
Convert DynamoDB decimals to floats

### LINKED ISSUE #5215: Tracker Serialization of Decimal Value
**Rasa version**: 1.5.1

**Issue**:
There are Decimal values associated with slots in our DynamoDB tracker store. When an action is run and the tracker store is json encoded it fails when attempting to serialize the Decimal value. This is similar to issue #2265

**Error (including full traceback)**:
```
2020-01-23 11:47:22 DEBUG    rasa.core.processor  - Object of type Decimal is not JSON serializable
Traceback (most recent call last):
File "/Users/dsammut/Repos/chatbot-rasa/venv/lib/python3.7/site-packages/rasa/core/processor.py", line 564, in _run_action
events = await action.run(output_channel, nlg, tracker, self.domain)
File "/Users/dsammut/Repos/chatbot-rasa/venv/lib/python3.7/site-packages/rasa/core/actions/action.py", line 541, in run
json=json_body, method="post", timeout=DEFAULT_REQUEST_TIMEOUT
File "/Users/dsammut/Repos/chatbot-rasa/venv/lib/python3.7/site-packages/rasa/utils/endpoints.py", line 146, in request
**kwargs,
File "/Users/dsammut/Repos/chatbot-rasa/venv/lib/python3.7/site-packages/aiohttp/client.py", line 1012, in __aenter__
self._resp = await self._coro
File "/Users/dsammut/Repos/chatbot-rasa/venv/lib/python3.7/site-packages/aiohttp/client.py", line 365, in _request
data = payload.JsonPayload(json, dumps=self._json_serialize)
File "/Users/dsammut/Repos/chatbot-rasa/venv/lib/python3.7/site-packages/aiohttp/payload.py", line 389, in __init__
dumps(value).encode(encoding),
File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/json/__init__.py", line 231, in dumps
return _default_encoder.encode(obj)
File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/json/encoder.py", line 199, in encode
chunks = self.iterencode(o, _one_shot=True)
File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/json/encoder.py", line 257, in iterencode
return _iterencode(o, 0)
File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/json/encoder.py", line 179, in default
raise TypeError(f'Object of type {o.__class__.__name__} '
TypeError: Object of type Decimal is not JSON serializable
```

**Content of configuration file (config.yml)** (if relevant):
```yml
language: en
# pretrained_embeddings_convert
pipeline:
- name: "WhitespaceTokenizer"
- name: "ConveRTFeaturizer"
- name: "EmbeddingIntentClassifier"
policies:
- name: FormPolicy
- name: MemoizationPolicy
- name: KerasPolicy
- name: MappingPolicy
- name: TwoStageFallbackPolicy
  nlu_threshold: 0.65
  core_threshold: 0.3
  fallback_core_action_name: action_fallback_transfer
  fallback_nlu_action_name: action_fallback_transfer
```

### PR DESCRIPTION
**Pull request by ricwo**
**Proposed changes**:
- fixes #5215 

**Status (please check what you already did)**:
- [x] added some tests for the functionality
- [ ] updated the documentation
- [x] updated the changelog (please check [changelog](https://github.com/RasaHQ/rasa/tree/master/changelog) for instructions)
- [x] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa#code-style) for instructions)

### CODE REVIEW

File: rasa/core/utils.py [hunk 1]
```diff
@@ -482,38 +482,61 @@ def handler(fut: Future) -> None:
...
+    def _float_to_rounded_decimal(f: float) -> Decimal:
+        return Decimal(f).quantize(Decimal(10) ** -round_digits)
+
+    return json.loads(json.dumps(obj), parse_float=_float_to_rounded_decimal)
+
+
+class DecimalEncoder(json.JSONEncoder):
+    """`json.JSONEncoder` that dumps `Decimal`s as `float`s."""
+
+    def default(self, o: Any) -> Any:
```
Comment (tabergma) [inline rasa/core/utils.py]: Can we rename `o` to `obj`?

File: rasa/core/tracker_store.py [hunk 2]
```diff
@@ -349,25 +349,31 @@ def serialise_tracker(self, tracker: "DialogueStateTracker") -> Dict:
...
-            )
-        else:
+
+        if not dialogues:
             return None
 
+        events = dialogues[0].get("events", [])
+
+        # `float`s are stored as `Decimal` objects - we need to convert them back
+        events_with_floats = utils.replace_decimals_with_floats(events)
```
Comment (tabergma) [inline rasa/core/tracker_store.py]: Minor: Should we import `utils` as `core_utils` to clarify where those methods come from?

Comment (ricwo) [reply rasa/core/tracker_store.py]: 👍

File: rasa/core/utils.py (hunk 1 above)
Comment (ricwo) [reply rasa/core/utils.py]: 👍

**Pull request merged**